.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,40 @@
+<details>
+<!-- inside this <details> section, markdown rendering does not work, so we have to use raw html here. -->
+<summary><h2> PR Checklist (Click to expand. Please do have a look before submitting a PR!) </h2></summary>
+<p>Before submitting your pull request, please ensure you have met the following criteria. This helps us maintain quality and efficiency in our review process.</p>
+
+<h3>PR Title And Classification</h3>
+<p><strong>I understand that</strong> only specific types of PRs will be reviewed. My PR title is prefixed appropriately to indicate the type of change, using one of the following:</p>
+<ul>
+    <li><code>[Doc]</code> for documentation fixes.</li>
+    <li><code>[Bugfix]</code> for bug fixes.</li>
+    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
+    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
+    <li><code>[Kernel]</code> for changes affecting computation kernels.</li>
+    <li><code>[Hardware][Vendor]</code> for hardware-specific changes. Vendor name should appear in the prefix, e.g., <code>[Hardware][AMD]</code>.</li>
+    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
+</ul>
+<p><strong>Note:</strong> If my PR spans more than one category, I will include all relevant prefixes.</p>
+
+<h3>Coding Standards</h3>
+<p><strong>I understand that</strong> I must run <code>./format.sh</code> <strong>before submitting the PR and after any new commits</strong> to ensure compliance with linter checks. PRs failing to meet linter standards will not be merged.</p>
+
+<h3>Code Quality</h3>
+<p><strong>I understand that</strong> my code must be well-commented and include sufficient tests, ensuring future contributors can easily understand and modify the codebase.</p>
```

Comment:
Can you add a check-box list here like the following:

- [ ] The PR is well-documented in the code, as well as in the document (`docs/source/`) if it adds new features.
- [ ] The PR adds unit tests for the new components.
- [ ] The PR adds integration tests for the new features.

Reply:
Also this check-box list should not be hided in the details.

Reply:
I use <details> here because @simon-mo wants PRs to look clean.

I think that is also reasonable, as these checklists are meant for pr submitters to read and check, they are not very meaningful to others.

Reply:
Overall we should trust the PR submitters. These checklists are more of **reminders** for them, rather than some items to be checked by the vllm team.

For a checkbox-like PR template, you can take a look at https://github.com/open-mmlab/mmcv/pull/3057 . Indeed it makes the PR filled with unmeaningful information. Collapsing them by default is better in my opinion.